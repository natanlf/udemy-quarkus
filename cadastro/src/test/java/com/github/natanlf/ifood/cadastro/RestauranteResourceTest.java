package com.github.natanlf.ifood.cadastro;

import static io.restassured.RestAssured.given;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.TestTransaction;
import io.quarkus.test.junit.QuarkusTest;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
//@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @TestTransaction
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes() {
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }

}