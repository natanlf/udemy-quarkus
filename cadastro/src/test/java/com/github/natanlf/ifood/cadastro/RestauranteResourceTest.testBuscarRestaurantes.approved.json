[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do keyclock"
  },
  {
    "id": 234,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do keyclock 2"
  },
  {
    "id": 333,
    "nome": "Novo Restaurante 4",
    "proprietario": "ID do keyclock 4"
  },
  {
    "id": 444,
    "nome": "Novo Restaurante 44",
    "proprietario": "ID do keyclock 44"
  }
]